file.go

```diff
@@ -37,29 +37,36 @@ func (f *file) Close() error {
 
 var _ fs.ReadDirFile = &file{}
 
-func (f *file) ReadDir(n int) ([]fs.DirEntry, error) {
+func (f *file) ReadDir(n int) (entries []fs.DirEntry, err error) {
 	if !f.IsDir() {
 		return nil, newErrNotDir("readdir", f.Name())
 	}
 
-	if n <= 0 {
-		return f.entries, nil
+	// directory already exhausted
+	if n <= 0 && f.readDirPos >= len(f.entries) {
+		return nil, nil
 	}
 
-	if f.readDirPos == len(f.entries) {
-		return nil, io.EOF
+	// read till end
+	if n > 0 && f.readDirPos+n > len(f.entries) {
+		err = io.EOF
```

Comment:
I just double checked how the `os` implementation behaves and it seems that we should never return a non empty slice and EOF at the same time.

For example with 3 files and n=2, I get:
f.ReadDir(2) // returns []{file1, file2}, nil
f.ReadDir(2) // returns []{file3}, nil
f.ReadDir(2) // returns []{}, EOF

So I think you can group the first 2 tests like this:
```go
if f.readDirPos >= len(f.entries) {
	if n <= 0 {
		return nil, nil
	} else {
		return nil, io.EOF
	}
}
```

It should allow you to get rid of the named return parameters, which is better for readability.

Reply:
Good point, I changed it.